### DISCUSSION TITLE
Typescript problem with registering a (non static) field name in V7

### DISCUSSION THREAD

**Discussion by medihack** [react-hook-form/react-hook-form#4536] [Q&A]
In a [Blitz.js](https://blitzjs.com/) I chose `react-hook-form` as the form library (it has some setup tool to let one choose). But after upgrading to `react-hook-form` V7 RC7 (and adapting some files) I get the following error when running `tsc`:

```
app/core/components/LabeledTextField.tsx:28:66 - error TS2345: Argument of type 'string' is not assignable to parameter of type '`${string}` | `${string}.${string}` | `${string}.${number}`'.

28           <input disabled={isSubmitting} {...props} {...register(props.name)} />
```

If I just do `register("foo")` typescript does not complain. I am pretty sure this wasn't the case with an earlier release candidate (RC3 or so).

Here is a repository that demonstrates the problem: https://github.com/medihack/react-hook-form-v7
To show the problem simply do `yarn; yarn run tsc`.

**Comment by bluebill1049** [comment]
`string` type is not going to satisfied with type check against the `name`, you will need to extend `Path<FormValue>`

**Comment by medihack** [comment]
Ok, then the error message is a bit hard to grasp (but I am a Typescript newbie). I also can't find anything about `Path` in the [V7 TS documentation](https://react-hook-form-website-git-v7-bluebill1049.vercel.app/ts). Is there some little example of how to type it correctly?

What is even more cumbersome for me is that I only get the error when running `tsc`, but not directly in Visual Studio Code (there it is fully ok that it is a string as `register` is typed like `const register: <string>(name: string, ....`).

**Reply by thenriquedb** [reply]
Hello, I am facing the same problem. How did you solve this? Did you have to switch to `useFieldArray`?

**Reply by medihack** [reply]
No, I simply ignore the name type in this particular form by using for example

    name={`${myVar1}.${myVar2}` as any}

EDIT: I do this also in v7 final release.

**Reply by bluebill1049** [reply]
> Hello, I am facing the same problem. How did you solve this? Did you have to switch to `useFieldArray`?

take a look at the typescript example for fieldarray, we have couple examples there.

**Comment by bluebill1049** [comment]
have a look at this example: https://codesandbox.io/s/react-hook-form-adapting-existing-form-ts-uzfxm?file=/src/index.tsx

**Reply by medihack** [reply]
Thanks for the help. This makes sense. I guess in v7 it is required that all possible form values must be known in advance when using types. The app I build has a very form-heavy plugin architecture. Form fields are added by dynamically imported components. I guess in this scenario I must give up the strongly typed benefits of react-hook-form, or is there a way to still retain some of the typings?
The only thing I know upfront when using `useForm` is this: `type FormValues={ [moduleId: string]: { [fielded: string]: any } }`

**Reply by bluebill1049** [reply]
Yes, @medihack hook form v7 is a strongly typed form. Right now dynamic field only applies to `useFieldArray`.
